### PR TITLE
adding target endpoint

### PR DESCRIPTION
**Pull request by harsha89**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: apim-operator/pkg/apis/wso2/v1alpha1/targetendpoint_types.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+type EndpointSecurity struct {
+	Username    string	`json:"username"`
+	Password    string  `json:"password"`
+	Type     	string  `json:"type"`
+}
+
+type Deploy struct {
+	Name 		   string	`json:"name"`
+	DockerImage    string	`json:"dockerImage"`
+	Count    	   int32    `json:"count"`
```
Comment (dinusha92) [inline apim-operator/pkg/apis/wso2/v1alpha1/targetendpoint_types.go]: Formatting issues in the code

File: apim-operator/pkg/apis/wso2/v1alpha1/targetendpoint_types.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+// +k8s:openapi-gen=true
+type TargetEndpointSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+	Type         		string 				`json:"type"`
+	Protocol     		string 				`json:"protocol"`
+	Hostname     		string 				`json:"hostname"`
+	Port         		int32  				`json:"port"`
+	Deploy  		    Deploy 				`json:"deploy"`
```
Comment (dinusha92) [inline apim-operator/pkg/apis/wso2/v1alpha1/targetendpoint_types.go]: Formatting issue

File: apim-operator/pkg/apis/wso2/v1alpha1/targetendpoint_types.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+}
+
+
+type EndpointSecurity struct {
+	Username    string	`json:"username"`
+	Password    string  `json:"password"`
+	Type     	string  `json:"type"`
```
Comment (dinusha92) [inline apim-operator/pkg/apis/wso2/v1alpha1/targetendpoint_types.go]: Formatting issue

File: apim-operator/pkg/apis/wso2/v1alpha1/targetendpoint_types.go (hunk 1 above)
Comment (harsha89) [reply apim-operator/pkg/apis/wso2/v1alpha1/targetendpoint_types.go]: Fixed with ead3e80eaf9bffe50f188f4b629ae9982370404c

File: apim-operator/pkg/apis/wso2/v1alpha1/targetendpoint_types.go (hunk 2 above)
Comment (harsha89) [reply apim-operator/pkg/apis/wso2/v1alpha1/targetendpoint_types.go]: Fixed with ead3e80eaf9bffe50f188f4b629ae9982370404c

File: apim-operator/pkg/apis/wso2/v1alpha1/targetendpoint_types.go (hunk 3 above)
Comment (harsha89) [reply apim-operator/pkg/apis/wso2/v1alpha1/targetendpoint_types.go]: Fixed with ead3e80eaf9bffe50f188f4b629ae9982370404c

**Pull request merged**